connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -763,6 +789,30 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
```

Comment:
My comment was a stretch. I made it because I saw the worker's executor being called only once when we `submit` a task during `startTask`. I oversaw the fact that it might be a longer running task.

Having said that, if we want to utilize the implicit barriers that are used by calling `invokeAll` or even `shutdown/awaitTermination` it's probably a better idea to keep these thread pools separated. One pool for start/stops and one for longer running tasks.